Rate limit desktop notifications

jcollie opened the pull request:
Add a once-per-second rate limit for all desktop notifications and a once-per-five-seconds rate limit for duplicate desktop notifications.

Fixes #1759.

jcollie committed:
suppress identical desktop notifications

jcollie committed:
add strict rate limit

jcollie committed:
adjust log message

mitchellh reviewed (changes requested):
One request, I also agree we should use a non-cryptographic hash and Wyhash is the typical one we use.

jcollie committed:
Address review comments

1. Switch to using Wyhash instead of a cryptographic hash.
2. Move global variables to App struct.

mitchellh committed:
core: nitpick some var names
